src/nvim/drawscreen.c

```diff
@@ -681,6 +683,39 @@ int update_screen(int type)
   return OK;
 }
 
+static int get_title_texts_len(VirtText title_texts)
+{
+  int len = 0;
+  for (size_t i = 0; i < title_texts.size; i++) {
+    char *text = title_texts.items[i].text;
+    len += (int)strlen(text);
+  }
+  return len;
+}
+
+static void redr_title_texts(win_T *wp, ScreenGrid *grid, int col)
+{
+  char *text;
+  int len, attr;
```

Comment:
You don't have to create forward declarations.
`int len = ...;` when it's actually needed is fine.

Reply:
oh just want this variable can be reused in a for loop. instead of creating a new temporary variable each time

Reply:
It's fine, the compiler will figure out this stuff.